### PR TITLE
[CrossSectionWeight] add Variations

### PR DESCRIPTION
**Pull request by cverstege**

### CODE REVIEW

File: KappaAnalysis/interface/KappaSettings.h [hunk 1]
```diff
@@ -102,6 +102,8 @@ class KappaSettings: public SettingsBase {
 
 
 	IMPL_SETTING(float, CrossSection);
+    IMPL_SETTING_DEFAULT(float, CrossSectionUp, 0.0)
+    IMPL_SETTING_DEFAULT(float, CrossSectionDown, 0.0)
```
Comment (HerrHorizontal) [inline KappaAnalysis/interface/KappaSettings.h]: I would recommend to use default values, which are not within the scope of the numerical reachable. Better use something negative, e.g. `-1`.

File: KappaAnalysis/src/Producers/CrossSectionWeightProducer.cc [hunk 2]
```diff
@@ -9,9 +9,14 @@ void CrossSectionWeightProducer::Produce(KappaEvent const &event, KappaProduct &
     assert(event.m_genLumiInfo);
     LOG(DEBUG) << "\n[" << this->GetProducerId() << "]";
 
-    if (static_cast<double>(settings.GetCrossSection()) > 0.0) {
+    if (settings.GetCrossSection() > 0.0) {
```
Comment (HerrHorizontal) [inline KappaAnalysis/src/Producers/CrossSectionWeightProducer.cc]: Would also change the check here...

File: KappaAnalysis/src/Producers/CrossSectionWeightProducer.cc [hunk 3]
```diff
@@ -9,9 +9,14 @@ void CrossSectionWeightProducer::Produce(KappaEvent const &event, KappaProduct &
...
         LOG(DEBUG) << "Adding crossSectionPerEventWeight: " << settings.GetCrossSection();
         product.m_weights["crossSectionPerEventWeight"] = settings.GetCrossSection();
+        if (settings.GetCrossSectionUp() > 0.0 && settings.GetCrossSectionDown() > 0.0) {
```
Comment (HerrHorizontal) [inline KappaAnalysis/src/Producers/CrossSectionWeightProducer.cc]: ... and here.

**Review by HerrHorizontal — CHANGES_REQUESTED** [review]

File: KappaAnalysis/interface/KappaSettings.h (hunk 1 above)
Comment (cverstege) [reply KappaAnalysis/interface/KappaSettings.h]: Ok, I'll change it to -1, but I think the check > 0 should stay, as the cross section should be positive.

**Review by HerrHorizontal — APPROVED** [review]
Looks good. Thank you

File: KappaAnalysis/interface/KappaSettings.h [hunk 4]
```diff
@@ -102,6 +102,8 @@ class KappaSettings: public SettingsBase {
 
 
 	IMPL_SETTING(float, CrossSection);
```
Comment (RHofsaess) [inline KappaAnalysis/interface/KappaSettings.h]: May add a default value of -1 here, too?
Else, this may brake if the CrossSection setting is not set.

Comment (cverstege) [reply KappaAnalysis/interface/KappaSettings.h]: I think the cross section should always be set. Shouldn't it?

Comment (HerrHorizontal) [reply KappaAnalysis/interface/KappaSettings.h]: Yes, I think it wouldn't make sense to run without giving a particular XS value. Is an error thrown, when the XS-value is not configured?

Comment (RHofsaess) [reply KappaAnalysis/interface/KappaSettings.h]: Yes it should.
At the moment, we have `cfg['CrossSection'] = -1  # need to be set in individual config file` in the default config.
I think, a settings default would be more reasonable :D
(also for the planned default config cleanup)

Comment (cverstege) [reply KappaAnalysis/interface/KappaSettings.h]: All right, I'll update it.

**Review by RHofsaess — APPROVED** [review]

**Pull request merged**